### ISSUE TITLE
CI / Deployment not working

### ISSUE THREAD

**Issue by ivanz** [dotnet/csharp-textmate-grammar#2]
@DustinCampbell Are you able to enable Travis CI for this repository? Otherwise no CI/deployment bits will work in `.travis.yml` I think.

**Comment by DustinCampbell** [comment]
Right. I'll need to figure out something for this.

**Comment by DustinCampbell** [comment]
The CI / Deployment we were using before was part of the OmniSharp organization. I'll need to look into getting something up and running for this.

**Comment by mmitche** [comment]
@DustinCampbell Let me know if you want to be hosted on .NET CI

**Comment by DustinCampbell** [comment]
Thanks for the @mmitche! @ivanz: would you be OK if we used Jenkins?

**Comment by ivanz** [comment]
Yeah should be fine as long as it can run NodeJS. Unless you already have a Dotnet org NPM account that can publish to NPM - it would also need to have a way to store/use encrypted token for `npm publish` (also email of the user).

**Comment by mmitche** [comment]
@ivanz We don't currently publish packages from the main CI instances (for security reasons).  There is a plan to bring up parallel instances in some form for these kind of scenarios.

However, we can at least get the builds and such in place.

**Comment by ivanz** [comment]
Fair enough. For CI I think all we need is `npm install && npm test` on a fresh checkout and getting it hooked into GitHub pull requests and commits to branches.

**Comment by DustinCampbell** [comment]
I wonder if it would make sense to go with Travis for simplicity? I'm not sure we need a lot of the machinery used for the .NET repos. Hmmm...

**Comment by ivanz** [comment]
Travis will certainly be the simplest if you can authorize it for this repo.

**Comment by mmitche** [comment]
Okay let me know if you need it or see long wait times on travis or whatnot.  Can you safely publish from travis?

**Comment by mmitche** [comment]
Also FYI the .NET CI setup is pretty simple too (no real knowledge of Jenkins necessary)

**Comment by JoeRobich** [comment]
Closing as this repository uses GitHub actions
